### DISCUSSION TITLE
Filtering

### DISCUSSION THREAD

**Discussion by waszpaw** [tommyblue/smugmug-backup#52] [Ideas]
Hi there. First of all - thanks for the tool, that's exactly what I was looking for, appreciated.
Are you thinking about enhancements? The only thing that I see missing (from my perspective of course) is possibility of choosing what I want to download (e.g. a wildcard for the album names or anything similar).
I might be trying to add it myself but since I have not touched Go before, it can be a bit challenging ;)
Cheers!
Pawel

**Comment by tommyblue** [comment]
Hi @waszpaw, I'm open to new features. I can try to make it in my spare time (not much with 2 little girls though 😅 ). Please write down some specs and I'll do my best

**Reply by waszpaw** [reply]
Super!

Let me try (although "specs" is a bit an overkill to describe):

As a user, I'd like to choose which folder /  gallery group is to be downloaded, rather than have them all downloaded / checked in one run. E.g. if the folders are organized with a year prefix (like 2023.10.10 XXX and 2022.12.05 YYY or 2023/ALL_GALLERIES_FROM_THIS_YEAR), I need to indicate that only these starting with "2023" are to be touched.

* To add a new configuration identifier, e.g. "store.mask". Required "no". Default value "*". If not NULL, it can have a wildcard form, e.g. "2023*". 
* Wildcard can have a form of starting or trailing or both, e.g. "*2023*", "2023*", "*GARDEN".
* Only folders / galleries with matching _full path_ will be considered (mind - it's not about the gallery name, rather the full string matching the pattern).
* All the other behavior stays the same (e.g. do not download something that is already here).
* To add a corresponding environmental variable (e.g. "SMGMG_BK_MASK") interpretation.

I (personally) don't need differentiation of download destinations or any other parameters depending on the mask. Maybe that's something for the future (if anybody is interested).
The most important is the _trailing_ wildcard, leading is not that crucial (e.g. downloading with PREFIX is a typica task, downloading with a searching key - a wish).

@tommyblue - please, let me know if that's understood, if not - I'll try to make it better :)
Cheers and thanks!
Pawel

**Reply by tommyblue** [reply]
@waszpaw ehm, sorry for the latency 😅 
Specs are clear, not sure if I'll have time for them though.
I recently started working to an [Electron version of this app](https://github.com/tommyblue/smugmug-backup-electron), maybe I could do something there (but still, without any imaginable ETA)
